presto-array/src/main/java/com/facebook/presto/array/Arrays.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.array;
+
+public class Arrays
+{
+    private Arrays() {}
+
+    public static int[] ensureCapacity(int[] buffer, int capacity)
+    {
+        return ensureCapacity(buffer, capacity, 1.0f, false, false);
+    }
+
+    public static int[] ensureCapacity(int[] buffer, int capacity, float expansionFactor, boolean needsCopy, boolean needsInitialization)
```

Comment:
@mbasmanova  Thanks for the suggestion, renamed them. For the expansion factor, I was thinking we could expose a knob on it so that we can experiment the best configuration later. I was thinking of creating a configuration property for it, does it sound like an overkill?

Reply:
@yingsu00 I think it makes sense to have a small number of well-defined policies that determine the expansion factor and use an enum to define these. I'd like to avoid having expansion factors hard-coded in various places in the code.

Reply:
@mbasmanova  It won't be hard coded, but will be replaced by a configuration property so that we can easily change the values in different runs, and its default value could be 1.0. How can we change this values without using a configuration property but just a set of predefined values?

Reply:
@mbasmanova My original idea of having expansionFactor is to making it configurable, e.g. make it a session/config property, so that we can easily experiment different values to get the best choice. After figuring out the best value, we can update the default config property. But Arrays.java only contains static methods and therefore it's not a good place to introduce the session/config property. The callers will have to get this session/config property by themselves. This seems a wider change so I decide not to do it in this PR. For now I introduced 3 enum values in Arrays.java: SMALL(1.0), MEDIUM(1.5) and LARGE(2.0) and callers would pass it in as a parameter.